src/pip/_internal/models/link.py

```diff
@@ -40,14 +52,15 @@ class Link(KeyBasedCompareMixin):
     def __init__(
         self,
         url: str,
-        comes_from: Optional[Union[str, "HTMLPage"]] = None,
+        comes_from: Optional[Union[str, "IndexContent"]] = None,
         requires_python: Optional[str] = None,
         yanked_reason: Optional[str] = None,
         cache_link_parsing: bool = True,
+        hashes: Optional[Mapping[str, str]] = None,
```

Comment:
Should this new param be mentoined in the docstring and and should we have a test for it ?

Reply:
Added it to the doc string, I don't see any tests for this class at all currently, but it should be tested implicitly in any of the tests that hit PyPI now.